### PR TITLE
Adding tool to check for new datasets in pinot and onboard them to thirdeye automatically

### PR DESCRIPTION
**Pull request by npawar**
The service will run periodically as a mode of thirdeye-cnotroller. It will add any new datasets it sees in pinot. It will also check all existing datasets to see if there are any changes to the schema

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/ThirdEyeAnomalyApplication.java [hunk 1]
```diff
@@ -94,6 +97,10 @@ public void start() throws Exception {
               new AnomalyMergeExecutor(executorService);
           anomalyMergeExecutor.start();
         }
+        if (config.isOnboard()) {
+          onboardPinotMetricsService = new OnboardPinotMetricsService(config);
```
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/ThirdEyeAnomalyApplication.java]: Onboard -> AutoLoad or AutoImport

Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/ThirdEyeAnomalyApplication.java]: change all variables accordingly

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/ConfigGenerator.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import com.linkedin.thirdeye.api.MetricType;
+import com.linkedin.thirdeye.datalayer.dto.DashboardConfigDTO;
+import com.linkedin.thirdeye.datalayer.dto.DatasetConfigDTO;
+import com.linkedin.thirdeye.datalayer.dto.IngraphMetricConfigDTO;
+import com.linkedin.thirdeye.datalayer.dto.MetricConfigDTO;
+import com.linkedin.thirdeye.datalayer.pojo.DashboardConfigBean;
+
+public class ConfigGenerator {
+
+  private static final String DEFAULT_METRIC_NAMES_COLUMN = "metricAlias";
```
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/ConfigGenerator.java]: this is for ingraph tables only right?

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/ConfigGenerator.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    // Create DatasetConfig
+    DatasetConfigDTO datasetConfigDTO = new DatasetConfigDTO();
+    datasetConfigDTO.setDataset(dataset);
+    datasetConfigDTO.setDimensions(dimensions);
+    datasetConfigDTO.setTimeColumn(timeSpec.getName());
+    datasetConfigDTO.setTimeDuration(timeSpec.getTimeUnitSize());
+    datasetConfigDTO.setTimeUnit(timeSpec.getTimeType());
+    datasetConfigDTO.setTimeFormat(timeSpec.getTimeFormat());
+    if (timeSpec.getTimeFormat().startsWith(TimeFormat.SIMPLE_DATE_FORMAT.toString())) {
+      datasetConfigDTO.setTimezone("US/Pacific");
```
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/ConfigGenerator.java]: use some constant or US/Pacific

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/OnboardPinotMetricsService.java [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+        }
+      }
+      refreshCaches();
+
+    } catch (IOException e) {
+      LOG.error("Exception in loading datasets", e);
+    }
+  }
+
+  public boolean isIngraphDataset(String dataset) {
```
Comment (kishoreg) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/OnboardPinotMetricsService.java]: what if we have created table but not added any metrics? Its better to read the schema and look for some specific columns in the schema

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/ConfigGenerator.java (hunk 3 above)
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/ConfigGenerator.java]: fixed

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/ConfigGenerator.java (hunk 2 above)
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/ConfigGenerator.java]: fixed

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/OnboardPinotMetricsService.java (hunk 4 above)
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/onboard/pinot/metrics/OnboardPinotMetricsService.java]: Changed it to check for metricAlias and value

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/ThirdEyeAnomalyApplication.java (hunk 1 above)
Comment (npawar) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/ThirdEyeAnomalyApplication.java]: fixed

**Pull request merged**